contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -190,15 +220,27 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
     }
 
     /**
-     * @dev See {acceptDefaultAdminTransfer}.
+     * @dev See {scheduleDefaultAdminDelayChange}.
      *
      * Internal function without access restriction.
      */
-    function _acceptDefaultAdminTransfer() internal virtual {
-        require(_hasDefaultAdminTransferDelayPassed(), "AccessControl: transfer delay not passed");
-        _revokeRole(DEFAULT_ADMIN_ROLE, defaultAdmin());
-        _grantRole(DEFAULT_ADMIN_ROLE, pendingDefaultAdmin());
-        _resetDefaultAdminTransfer();
+    function _scheduleDefaultAdminDelayChange(uint48 newDefaultAdminDelay) internal virtual {
+        require(defaultAdminTransferSchedule() != 0, "AccessControl: default admin transfer pending");
+
+        uint48 currentDefaultAdminDelay = defaultAdminDelay();
+
+        // Schedules the current delay if the delay is increased.
+        // Otherwise, wait the difference between current and new delay to guarantee the delay change schedule + a default admin change
+        // is effectively the current delay. For example, if delay is reduced from 10 days to 3 days, it's needed to wait 7 days
+        // before starting the new 3 days delayed transfer suming up to 10 days, which is the current delay.
+        uint48 changeDelay = newDefaultAdminDelay >= currentDefaultAdminDelay
+            ? newDefaultAdminDelay
+            : currentDefaultAdminDelay - newDefaultAdminDelay; // This is 1 in worst-case scenario. Can't be 0.
```

Comment:
What does it matter if it's 0?

Reply:
> Why does it matter if it's 0? 

It matters because we don't know what's going to be used for. Since there's a `pendingDefaultAdminDelay()` getter, someone implementing on top of this may rely on that not being 0.

I'd rather keep it in 1 at least to avoid accidental allowance of bypassing the `block.timestamp` in the same transaction.

Let me know if this shouldn't even be a concern.

Reply:
After discussing, we decided to tolerate the 0.